compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -1842,7 +1842,24 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             expr_t
         );
         err.span_label(field.span, "method, not a field");
-        if !self.expr_in_place(expr.hir_id) {
+        let expr_is_call =
+            if let hir::Node::Expr(hir::Expr { kind: ExprKind::Call(callee, _args), .. }) =
+                self.tcx.hir().get(self.tcx.hir().get_parent_node(expr.hir_id))
+            {
+                expr.hir_id == callee.hir_id
+            } else {
+                false
+            };
+        let expr_snippet =
+            self.tcx.sess.source_map().span_to_snippet(expr.span).unwrap_or(String::new());
+        if expr_is_call && expr_snippet.starts_with("(") && expr_snippet.ends_with(")") {
+            err.span_suggestion_short(
+                expr.span,
+                "remove wrapping parentheses to call the method",
+                expr_snippet[1..expr_snippet.len() - 1].to_owned(),
+                Applicability::MachineApplicable,
+            );
```

Comment:
(The AST holds an `ast::ExprKind::Paren`, but the HIR doesn't).

Ideally we would give a `multipart_suggestion` where the suggestion is something like

```
let suggestion = vec![
    (expr.span.with_hi(callee.span.lo()), String::new()),
    (expr.span.with_lo(callee.span.hi()), String::new()),
];
```